### PR TITLE
Configure TracerFilter only in Spring MVC environments

### PR DESCRIPTION
**Pull request by jrehwaldt**
Extract `TracerFilter` configuration into dedicated `TracerWebMvcAutoConfiguration` class, which is only loaded for active Spring MVC aka Servlet environments.

## Motivation and Context
Fixes gh-154

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: tracer-spring-boot-starter/src/main/java/org/zalando/tracer/spring/TracerWebMvcAutoConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+@API(status = STABLE)
+@Configuration
+@ConditionalOnClass({Tracer.class, Filter.class, FilterRegistrationBean.class})
+@ConditionalOnWebApplication
+@AutoConfigureAfter(name = {
+        "org.springframework.boot.autoconfigure.web.WebMvcAutoConfiguration", // Spring Boot 1.x
+        "org.springframework.boot.autoconfigure.web.servlet.WebMvcAutoConfiguration" // Spring Boot 2.x
+})
+@Import({TracerAutoConfiguration.class})
+public class TracerWebMvcAutoConfiguration {
```
Comment (whiskeysierra) [inline tracer-spring-boot-starter/src/main/java/org/zalando/tracer/spring/TracerWebMvcAutoConfiguration.java]: This class needs to be a static inner class of `TracerAutoConfiguration`.

Comment (jrehwaldt) [reply tracer-spring-boot-starter/src/main/java/org/zalando/tracer/spring/TracerWebMvcAutoConfiguration.java]: I pushed the requested changes. Proper testing is not possible using this setup so I reverted to the initial tests.

**Comment by whiskeysierra** [comment]
Thanks for your contribution! :tada:

**Comment by whiskeysierra** [comment]
I deployed the fix as 0.17.1 to central.

**Pull request merged**